design/ceph-config-updates.md

```diff
@@ -0,0 +1,127 @@
+containerized application space than before.
+
+However, for it's own backwards compatibility, Ceph options set in the config file or via command
+line flags will override the centrally-configured settings. To make the most of this functionality
+within Ceph, it will be necessary to limit the configuration options specified by Rook in either
+config files or on the command line to a minimum.
+
+## Proposed new design
+
+The end goal of this work will be to allow Ceph to centrally manage its own configuration as much as
+possible. Or in other terms, Rook will specify the barest minimum of configuration options in the
+config file or on the command line with a priority on cleaning the config file.
+
+Rook currently offers the option of a config override, and this functionality can be kept for
+backwards compatibility for existing users; however, we should note that it is advisable not to do
+this and instead configure Ceph's centrally-managed options.
+
+### Minimal config file
+
+#### Required minimum
+All Ceph options in the config file can be set via the command line, so it is tempting to remove the
+need for having a `ceph.conf` in containers at all. In practice, we do not know how to make this an
+actual possibility.
+
+The troublesome situation regards Ceph daemons' ability to discover mons. The daemons must have
+`mon host` specified to find mons which are using non-default ports. It is further undesirable to
+specify this via the `--mon-host` flag because if this parameter is specified statically as a
+container argument and the initial mons fail and are replaced by new mons, the daemon will not know
+how to contact the mons if it is restarted.
+
+The proposed solution to this problem and the only config option which have thus far been deemed
+*required* is the `mon host` global option which will be stored in a Kubernetes ConfigMap so that
+containers always have the most updated version of this information. The single `ceph.conf`
+ConfigMap will be mounted to all containers at `/etc/ceph/ceph.conf`. There is a possibility the
+`mon initial members` global option may also be necessary, as it goes hand-in-hand with `mon host`.
+
+With this one -- or these two -- options set via the config file, all daemons should *require* no
+other config file options set.
+
+#### Convenient minimum
+
+There may be a small set of additional options which are okay to have in the config file for
+convenience if they would otherwise be set via command line for every single daemons.
+
+For example:
+- `log file = /dev/stderr` should be set for all daemons to enforce the container standard of
+  sending logging to stdout/stderr.
+- `mon cluster log file = /dev/stderr` also for reasons above
+- `log stderr prefix = "debug "` should be set for all daemons to differentiate logging from auditing
+  - TODO: should this be set via Ceph centralized config if someone wants to change this? What about Luminous?
```

Comment:
It is better to set these via the mon config or else they cannot be easily overridden (e.g., by some poor admin trying to debug a problem).  IMO the ideal sequence would be for rook, during cluster creation, to form the initial mon quorum, and then inject a bunch of settings it wants (including the 'config overrides', probably) via 'ceph config set ...' or even 'ceph config assimilage-conf' to slurp up a bunch in one step (faster!), the proceed to start the rest of the cluster.

Reply:
I am inclined to keep the log files explicitly set to `/dev/stderr`. This is a standard in containers, and while a sysadmin might wish to change the log location for their own debugging, this would also mean that logs might not get picked up by fluentd or whatever log aggregator, which could be a worse issue. I'm not a sysadmin, however, so I may not have a good understanding of the user here. Perhaps @druonysus has a thought?

Reply:
@BlaineEXE I think your instinct to stick with the standard is a good one. No sense in us going against it. Keep it this why by default but make it configurable  to change, if needed. I know there are people who will want to change this.